zilliqa/src/db.rs

```diff
@@ -1166,45 +1216,12 @@ pub struct TrieStorage {
     cache: Arc<Mutex<LruCache<Vec<u8>, Vec<u8>>>>,
 }
 
-impl eth_trie::DB for TrieStorage {
-    type Error = rusqlite::Error;
-
-    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
-        if let Some(cached) = self.cache.lock().unwrap().get(key).map(|v| v.to_vec()) {
-            return Ok(Some(cached));
-        }
-
-        let value: Option<Vec<u8>> = self
-            .db
-            .lock()
-            .unwrap()
-            .query_row(
-                "SELECT value FROM state_trie WHERE key = ?1",
-                [key],
-                |row| row.get(0),
-            )
-            .optional()?;
-
-        let mut cache = self.cache.lock().unwrap();
-        if !cache.contains(key) {
-            if let Some(value) = &value {
-                let _ = cache.insert(key.to_vec(), value.clone());
-            }
-        }
-
-        Ok(value)
-    }
-
-    fn insert(&self, key: &[u8], value: Vec<u8>) -> Result<(), Self::Error> {
-        self.db.lock().unwrap().execute(
-            "INSERT OR REPLACE INTO state_trie (key, value) VALUES (?1, ?2)",
-            (key, &value),
-        )?;
-        let _ = self.cache.lock().unwrap().insert(key.to_vec(), value);
-        Ok(())
-    }
-
-    fn insert_batch(&self, keys: Vec<Vec<u8>>, values: Vec<Vec<u8>>) -> Result<(), Self::Error> {
+impl TrieStorage {
+    pub fn write_batch(
```

Comment:
Maybe nitpicking - Is this called anywhere other than `insert_batch` in the `eth_trie::DB` implementation? Why do we need to separate it out and add an extra function call?